plugins/templates/create.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const hoek = require('hoek');
+const urlLib = require('url');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/templates',
+    config: {
+        description: 'Create a new template',
+        notes: 'Create a specific template',
+        tags: ['api', 'templates'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['build']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const templateFactory = request.server.app.templateFactory;
+            const token = request.auth.credentials;
+            const pipelineId = token.pipelineId;
+            const name = request.payload.name;
+            const version = request.payload.version;
+            const labels = request.payload.labels || [];
+
+            return Promise.all([
+                pipelineFactory.get(pipelineId),
+                templateFactory.list({ name })
+            ]).then(([pipeline, templates]) => {
+                const templateConfig = hoek.applyToDefaults(request.payload, {
+                    scmUri: pipeline.scmUri,
+                    labels
+                });
+
+                // If template doesn't exist yet, just create a new entry
+                if (templates.length === 0) {
+                    return templateFactory.create(templateConfig);
+                }
+
+                // If template exists, but this build's scmUri is not the same as template's scmUri
+                // Then this build does not have permission to publish
```

Comment:
We didn't come to a conclusion for that. I personally think that if the repo changes, it should be a different template. Otherwise, I don't know what to use for permission control. But we can talk about that more in the 2nd pass.